### PR TITLE
UI Cleanup and improvements

### PR DESCRIPTION
**Pull request by jobobby04**
Finished a bunch of UI tweaks. Main highlights are
- Padding for click indicators look much better by applying padding after the clickable. This is done by removing padding from parent composables and applying it to the children instead.
- Add animations where possible, like expanding filter menu groups or switching between tabs
- Images now all crossfade between states
- Better image loading error display
- Consistent image loading placeholder
- Add display for languages instead of just showing the language code
- The linear loading indicator now overlays instead of being added between the toolbar and the content
- Use OutlinedTextFields instead of basic TextFields since they look better
- Handle no more content in catelog properly so its not displayed before loading
- Add BackHandler for library selection so it doesnt close the app when trying to deselect
- Replace clickable unread chip with a badge thats not clickable since it doesnt make sense to click it
- Selection background instead of a border for library and chapter list
- Clicking a settings item now starts/toggles/expands that setting
- Clicking a filter item now toggles/expands that filter
- Use a better cover ratio

#### Ignore the theme colors these were taken on different devices
| Old | New |
| ------- | ------- |
|Screenshot_20220806_170509|Screenshot_20220806_170535|
|Screenshot_20220806_170436|Screenshot_20220806_170415|
|Screenshot_20220806_171718|Screenshot_20220806_171800|
|Screenshot_20220807_191340|Screenshot_20220807_191306|
|Screenshot_20220806_171529|Screenshot_20220806_171641|
Screenshot_20220806_171157|Screenshot_20220806_171045|
|Screenshot_20220806_172137|Screenshot_20220806_172110|
|Screenshot_20220806_171233|Screenshot_20220806_171314|
|Screenshot_20220806_171402|Screenshot_20220806_171432|

### CODE REVIEW

File: android/src/main/java/app/shosetsu/android/ui/library/LibraryController.kt [hunk 1]
```diff
@@ -134,18 +158,6 @@ class LibraryController
...
-								makeSnackBar(
-									resources!!.getQuantityString(
-										R.plurals.toast_unread_count,
-										item.unread,
-										item.unread
-									)
-								)?.show()
-							} catch (e: Resources.NotFoundException) {
-							}
-						},
```
Comment (Doomsdayrs) [inline android/src/main/java/app/shosetsu/android/ui/library/LibraryController.kt:148]: Why kill my toast??

File: android/src/main/java/app/shosetsu/android/view/compose/Selected.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package app.shosetsu.android.view.compose
+
```
Comment (Doomsdayrs) [inline android/src/main/java/app/shosetsu/android/view/compose/Selected.kt:2]: License?

File: android/src/main/java/app/shosetsu/android/view/compose/ImageLoadingError.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package app.shosetsu.android.view.compose
+
```
Comment (Doomsdayrs) [inline android/src/main/java/app/shosetsu/android/view/compose/ImageLoadingError.kt:2]: License?

**Review by Doomsdayrs — CHANGES_REQUESTED** [review]
Aside from being one large commit, just address what I commented please.

Thank you for your work

**Comment by jobobby04** [comment]
Comments addressed 👍

File: android/src/main/java/app/shosetsu/android/view/compose/NovelCard.kt [hunk 4]
```diff
@@ -144,18 +144,24 @@ fun NovelCardNormalContent(
 					}
 					.alpha(if (isPlaceholder) 0.0f else 1.0f)
 			)
-			Text(
-				title,
-				modifier = Modifier
+			Box(
+				Modifier
 					.align(Alignment.BottomCenter)
-					.placeholder(visible = isPlaceholder)
```
Comment (Doomsdayrs) [inline android/src/main/java/app/shosetsu/android/view/compose/NovelCard.kt:148]: This is meant for search views where content is loaded via paging. Why are you removing it?

Comment (jobobby04) [reply android/src/main/java/app/shosetsu/android/view/compose/NovelCard.kt:148]: I didn't, it's still on the Text, not the Box around the Text

**Comment by Doomsdayrs** [comment]
Alright, final two notes and I will be happy with this.

1. Undo the change of turning the update header from a card to a surface, it violates material guidelines regarding elevation and objects moving under each other.
2. The animation of language filters in browse filter menu causes a stuttery transition. Have any clue on this?

**Comment by jobobby04** [comment]
1. I'll add the elevation back, I think I won't re-add the rounded corners since that's what made it look horrible in the first place.
2. I'll take a look at it, though I'm not sure why it would cause that

**Comment by jobobby04** [comment]
Fixed 👍

**Review by Doomsdayrs — APPROVED** [review]

**Pull request merged**